mewarpx/mewarpx/assemblies.py

```diff
@@ -417,20 +434,33 @@ def __init__(self, V, T, WF, name, r_outer, r_inner=0.0):
 
         # check if the cylinder forms the simulation boundary
         if np.isclose(self.r_inner, mwxrun.rmax):
+            if not infinite_cyl:
+                raise AttributeError("CylinderZ outside simulation domain.")
```

Comment:
```suggestion
                raise AttributeError(
                    "CylinderZ on the simulation boundary must span the full "
                    "z domain."
                )
```
You can change the wording if you want - but notice your error text doesn't make a ton of sense in that if r_inner > mwxrun.rmax this error won't appear, and the error is not really that the r bound is necessarily wrong but rather that zmin/zmax are wrong for a cylinder on the boundary.